autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -246,13 +247,12 @@ def transform(
         # having a value for a categorical column.
         # We need to convert the column in test data to
         # object otherwise the test column is interpreted as float
-        if len(self.categorical_columns) > 0:
-            categorical_columns = self.column_transformer.transformers_[0][-1]
-            for column in categorical_columns:
-                if X[column].isna().all():
-                    X[column] = X[column].astype('object')
-
         if self.column_transformer is not None:
```

Comment:
Is this just to pass the tests, theoretically this is always instantiated during `_fit` right

Reply:
So, the lines moved into this condition that is to make columns 'object' dtype, require that the data `X` is a dataframe. If there is a column transformer,  we are sure that the data is a dataframe.

Reply:
we technically also support sparse data. So the test for sparse data were failing

Reply:
Ah, then it is understandable.